textract/parsers/__init__.py

```diff
@@ -34,9 +35,11 @@ def process(filename, encoding=DEFAULT_ENCODING, **kwargs):
     # get the filename extension, which is something like .docx for
     # example, and import the module dynamically using importlib. This
     # is a relative import so the name of the package is necessary
-    _, ext = os.path.splitext(filename)
-    ext = ext.lower()
-
+    if not extension:# if no extension passed, extract the extension from the file
+        _, ext = os.path.splitext(filename)
+        ext = ext.lower()
+    else:# use the passed extension (useful for files without extension)
+        ext = extension
```

Comment:
if you end up allowing extensions to be passed without the leading `.`, be sure to add it to the beginning of the `ext` variable. You would need to have something like:

```python
if not ext[0] == '.':
    ext = '.' + ext
```